### ISSUE TITLE
Doubts regarding for Kitti dataset

### ISSUE THREAD

**Issue by koutilya-pnvr** [lyndonzheng/Synthetic2Realistic#30]
Hi, I find the repo to be really useful. Thanks a lot for sharing firstly.

For Virtual Kitti -> Kitti translation, were you also making use of the left-right images together? Or were you just considering Left (or Right) images alone?
There were various files in the datasplit folder! Which of them were used as real dataset (Kitti) during training (or evaluating) in the original work?

Also on a different note what do you mean by the 'Dataset paired | Unpaired' command line option?

**Comment by koutilya-pnvr** [comment]
Can you share the input and label file lists for the Real dataset Kitti? Just like you shared the file lists for the Virtual Kitti in isssue #5 

There were GT depths already given by the [official Kitti webpage](http://www.cvlibs.net/datasets/kitti/eval_depth.php?benchmark=depth_prediction). Were you creating your own Ground truths for evaluating your model?

**Comment by lyndonzheng** [comment]
@koutilya40192 We use both left-right images together. These left-right images just be used to make the virtual kitti images like the real world images.

As for the evluation, we follow the [monodepth](https://github.com/mrharicot/monodepth/blob/master/utils/evaluate_kitti.py) to get the ground truth.

**Comment by koutilya-pnvr** [comment]
Thanks for your response. So, you just use all the left-right images to translate Virtual Kitti to yield realistic images. Were you using these left-right images in some order to make use of the stereo information or were you simply throwing all the Kitti images (in random order) to discriminator as real images?

Also what does the 'Dataset paired | Unpaired' command line option stand for? What does it do?

**Comment by lyndonzheng** [comment]
@koutilya40192 We do not use the stereo information for the depth predication. We just simple throw all Kitti Images to the discriminator. 

The "Paired|Unpaired" command tries to make the source and target image correspoding to each other on the content. This command is used to do the general image translation task, such as zebra2horse, apple2orange.

**Comment by koutilya-pnvr** [comment]
Thanks for clarification on the left-right images. 

I still dont understand why this 'Paired | Unpaired' option is present in your repo, when your entire training methodology is focused on Image translation using unpaired real images. Is the context of 'unpaired' in this repo different that of written in your abstract? If so what does the 'Unpaired|Paired' mean exactly here.

![image](https://user-images.githubusercontent.com/24998956/64664004-b6f18d80-d41b-11e9-8d7b-f25e70959100.png)

I understand you have just explained this, but i dont get the reference to your paper regarding 'Unpaired'.

In other words what did this work assume during training the T2Net? Paired or Unpaired command?

**Comment by koutilya-pnvr** [comment]
> @koutilya40192 We use both left-right images together. These left-right images just be used to make the virtual kitti images like the real world images.
> 
> As for the evluation, we follow the [monodepth](https://github.com/mrharicot/monodepth/blob/master/utils/evaluate_kitti.py) to get the ground truth.

I have tried the generate_depth_map() function and found that the ground truth depth for Kitti still seems to be sparse. 
https://github.com/lyndonzheng/Synthetic2Realistic/blob/f06db4734b89f80b76f9362ddec17f8eaf1216db/util/data_kitti.py#L175
https://github.com/lyndonzheng/Synthetic2Realistic/blob/f06db4734b89f80b76f9362ddec17f8eaf1216db/util/evaluation.py#L37
Do you tweak the interp=False to True during evaluation to get completed depth maps?

**Comment by lyndonzheng** [comment]
@koutilya40192 The unpaired images means during the training, we do not have the target realistic image for a given synthetic image. Hence, in our T2Net,, the depth predication task, we only use the "unpaired" command. The "paired" command is used for the [pix2pix](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix)

For the depth map, we use interp=True, so that the depth map is dense and can be used for training. However, for the testing, we do not consider the missing depth pixel. This is follow the other previous work.

**Comment by koutilya-pnvr** [comment]
Thanks for the clarification.
